src/contracts/core/RewardsCoordinator.sol

```diff
@@ -340,6 +349,83 @@ contract RewardsCoordinator is
         }
     }
 
+    /**
+     * @notice Validate a PerformanceRewardsSubmission. Called from `createAVSPerformanceRewardsSubmission`.
+     * @param performanceRewardsSubmission PerformanceRewardsSubmission to validate.
+     * @return total amount to be transferred from the avs to the contract.
+     */
+    function _validatePerformanceRewardsSubmission(
+        PerformanceRewardsSubmission calldata performanceRewardsSubmission
+    ) internal view returns (uint256) {
+        require(
+            performanceRewardsSubmission.strategiesAndMultipliers.length > 0,
+            "RewardsCoordinator._validatePerformanceRewardsSubmission: no strategies set"
+        );
+        require(
+            performanceRewardsSubmission.operatorRewards.length > 0,
+            "RewardsCoordinator._validatePerformanceRewardsSubmission: no operators rewarded"
+        );
+
+        uint256 totalAmount = 0;
+        address currOperatorAddress = address(0);
+        for (uint256 i = 0; i < performanceRewardsSubmission.operatorRewards.length; ++i) {
+            OperatorReward calldata operatorReward = performanceRewardsSubmission.operatorRewards[i];
+            require(
+                operatorReward.operator != address(0),
+                "RewardsCoordinator._validatePerformanceRewardsSubmission: operator cannot be 0 address"
+            );
+            require(
+                currOperatorAddress < operatorReward.operator,
+                "RewardsCoordinator._validatePerformanceRewardsSubmission: operators must be in ascending order to handle duplicates"
```

Comment:
Is this a more gas efficient way of not having a temporary storage slot? Seems rather restrictive on the client but not seemingly impossible I suppose.

Reply:
Also, who cares if there are duplicates? Is it up to us to infer their intention? Are there any side effects of allowing duplicates?

Reply:
Is your question: why are we doing this check? If that's the case we're doing it so that the AVS can't pass in duplicate operator addresses into the reward submission, and this is the cheapest way we can check that without doing any searching of the array. 

The cli/sdk can easily sort it into this order and warn if there are duplicates before the transaction is submitted onchain.

Reply:
We do this duplicate check for the Rewards v1 submission. So I'm assuming it's some constraint on the pipeline and to make some calculation easier (I'm just speculating, but I could be wrong - Need to check this assumption). 

But besides that why would an AVS need to pay the same operator 2 different amounts in the same reward submission? This seems like a nice constraint to have.